adapters/Pangle/PangleAdapter/GADMediationAdapterPangle.m

```diff
@@ -19,23 +19,31 @@
 #import "GADPangleRTBBannerRenderer.h"
 #import "GADPangleRTBInterstitialRenderer.h"
 #import "GADPangleRTBRewardedRenderer.h"
+#import "GADPangleRTBNativeRenderer.h"
+#import "GADPangleNetworkExtras.h"
 
 static NSInteger _gdpr = -1, _ccpa = -1;
 
 @implementation GADMediationAdapterPangle {
-  /// Pangle banner ad wrapper.
-  GADPangleRTBBannerRenderer *_bannerRenderer;
-  /// Pangle interstitial ad wrapper.
-  GADPangleRTBInterstitialRenderer *_interstitialRenderer;
-  /// Pangle rewarded ad wrapper.
-  GADPangleRTBRewardedRenderer *_rewardedRenderer;
+    /// Pangle banner ad wrapper.
+    GADPangleRTBBannerRenderer *_bannerRenderer;
+    /// Pangle interstitial ad wrapper.
+    GADPangleRTBInterstitialRenderer *_interstitialRenderer;
+    /// Pangle rewarded ad wrapper.
+    GADPangleRTBRewardedRenderer *_rewardedRenderer;
+    /// Pangle native ad wrapper.
+    GADPangleRTBNativeRenderer *_nativeRenderer;
 }
 
 - (void)collectSignalsForRequestParameters:(nonnull GADRTBRequestParameters *)params
                          completionHandler:
-                             (nonnull GADRTBSignalCompletionHandler)completionHandler {
-  NSString *signals = [BUAdSDKManager getBiddingToken:nil];
-  completionHandler(signals, nil);
+(nonnull GADRTBSignalCompletionHandler)completionHandler {
+    GADPangleNetworkExtras *extras = [params.extras isKindOfClass:[GADPangleNetworkExtras class]] ? params.extras : nil;
+    if (extras && extras.userDataString.length > 0) {
+        [BUAdSDKConfiguration configuration].userExtData = extras.userDataString;
+    }
```

Comment:
Why is this placed here? And when should the `userDataString` be passed to Pangle SDK?

Reply:
@wangyuanyuanyy to respond. Thanks!